play-services-basement/src/main/java/com/google/android/gms/dynamic/ObjectWrapper.java

```diff
@@ -34,14 +34,24 @@ public static Object unwrap(IObjectWrapper obj) {
         if (obj == null) {
             return null;
         }
+
         if (obj instanceof ObjectWrapper) {
             return ((ObjectWrapper) obj).t;
         }
+
         IBinder binder = obj.asBinder();
         Field[] fields = binder.getClass().getDeclaredFields();
-        if (fields.length != 1) {
-            throw new IllegalArgumentException();
+
+        if (fields.length < 1) {
+            throw new IllegalArgumentException("No fields were found");
         }
+
+        // Ignore synthetic field from JaCoCo
+        // https://www.jacoco.org/jacoco/trunk/doc/faq.html
+        if (fields.length > 2 || (fields.length == 2 && !fields[1].isSynthetic())) {
```

Comment:
I'm not sure you can rely on this field being added as a second field.

Reply:
Thank you for the review, that's a fair concern.  I've updated the handling to be order independent.

Reply:
What about just count not-sysnthetic fileds?
So:
- Synthetic fields can be any number even 100
- Not-synthetic fieds must be 1

Quote:
> Please change your code to ignore synthetic members. This is a good practice anyways as also the Java compiler creates synthetic members in certain situation.